### PR TITLE
DEV-4283: Export query

### PR DESCRIPTION
**Pull request by jkgenser**

### CODE REVIEW

**Comment by jkgenser** [comment]
Added Backend team to the PR so people can start getting familiar- but Nic has been the main code owner of the current python client.

File: indico/queries/export.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        super().__init__()
+
+    def requests(self):
+        yield _CreateExport(
+            dataset_id=self.dataset_id,
+            column_ids=self.column_ids,
+            subset_ids=self.subset_ids,
+            labelset_ids=self.labelset_ids,
+            file_info=self.file_info,
+            combine_labels=self.combine_labels,
```
Comment (pastap) [inline indico/queries/export.py:201]: I believe you're missing the anonymous kwarg here
and in the mutation above

Comment (jkgenser) [reply indico/queries/export.py:201]: will update.

File: indico/queries/export.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+    def __init__(self, export_id: int):
+        super().__init__(self.query, variables={"exportIds": [export_id]})
+
+    def process_response(self, response):
+        response = super().process_response(response)
+        return Export(**response["exports"]["exports"][0])
+
+
+class RetrieveExport(RetrieveStorageObject):
```
Comment (nkassis) [inline indico/queries/export.py:112]: Should this be _RetrieveExport? Do we expect people to call this directly?

File: indico/queries/export.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        self.export = export
+
+    def requests(self):
+        if self.export_id:
+            yield GetExport(self.export_id)
+        export = self.export or self.previous
+
+        if export.status != "COMPLETE":
+            raise IndicoRequestError(
+                code="JOB_INCOMPLETE",
```
[the remark this answers is not in the document]
Comment (jkgenser) [reply indico/queries/export.py]: updated

File: indico/queries/export.py (hunk 2 above)
Comment (jkgenser) [reply indico/queries/export.py]: good point. updated to RetrieveExport. DownloadExport is the public query

**Pull request merged**